in_toto/verifylib_test.go

```diff
@@ -62,16 +69,120 @@ func TestInTotoVerifyPass(t *testing.T) {
+		t.Errorf("Summary Link isn't of type Link")
+	}
+	if !reflect.DeepEqual(summaryLink.Signed.(Link).Name,
+		codeLink.Signed.(Link).Name) {
+		t.Errorf("Summary Link name doesn't match. Expected '%s', returned "+
+			"'%s", codeLink.Signed.(Link).Name, summaryLink.Signed.(Link).Name)
+	}
+	if !reflect.DeepEqual(summaryLink.Signed.(Link).Materials,
+		codeLink.Signed.(Link).Materials) {
+		t.Errorf("Summary Link materials don't match. Expected '%s', "+
+			"returned '%s", codeLink.Signed.(Link).Materials,
+			summaryLink.Signed.(Link).Materials)
+	}
+
+	if !reflect.DeepEqual(summaryLink.Signed.(Link).Products,
+		packageLink.Signed.(Link).Products) {
+		t.Errorf("Summary Link products don't match. Expected '%s', "+
+			"returned '%s", packageLink.Signed.(Link).Products,
+			summaryLink.Signed.(Link).Products)
+	}
+	if !reflect.DeepEqual(summaryLink.Signed.(Link).Command,
+		packageLink.Signed.(Link).Command) {
+		t.Errorf("Summary Link command doesn't match. Expected '%s', "+
+			"returned '%s", packageLink.Signed.(Link).Command,
+			summaryLink.Signed.(Link).Command)
+	}
+	if !reflect.DeepEqual(summaryLink.Signed.(Link).ByProducts,
+		packageLink.Signed.(Link).ByProducts) {
+		t.Errorf("Summary Link by-products don't match. Expected '%s', "+
+			"returned '%s", packageLink.Signed.(Link).ByProducts,
+			summaryLink.Signed.(Link).ByProducts)
+	}
+	if !reflect.DeepEqual(summaryLink.Signed.(Link).ByProducts["return-value"],
+		packageLink.Signed.(Link).ByProducts["return-value"]) {
+		t.Errorf("Summary Link return value doesn't match. Expected '%s', "+
+			"returned '%s", packageLink.Signed.(Link).ByProducts["return-value"],
+			summaryLink.Signed.(Link).ByProducts["return-value"])
+	}
 }
 
-func TestInTotoVerifyLayoutDoesNotExist(t *testing.T) {
-	err := InTotoVerify("layout/does/not/exist", map[string]Key{},
-		"link/dir/does/not/matter")
-	// Asssert error type to PathError
-	if _, ok := err.(*os.PathError); ok == false {
-		t.Fail()
+func TestVerifySublayouts(t *testing.T) {
+	sublayoutName := "sub_layout"
+	var aliceKey Key
+	if err := aliceKey.LoadPublicKey("alice.pub"); err != nil {
+		t.Errorf("Unable to load Alice's public key.")
+	}
+	sublayoutDirectory := fmt.Sprintf(SublayoutLinkDirFormat, sublayoutName,
+		aliceKey.KeyId)
+	if err := os.Mkdir(sublayoutDirectory, 0700); err != nil {
+		t.Errorf("Unable to create sublayout directory.")
+	}
+	writeCodePath := path.Join(sublayoutDirectory, "write-code.776a00e2.link")
+	if err := os.Link("write-code.776a00e2.link", writeCodePath); err != nil {
+		t.Errorf("Unable to link write-code metadata.")
+	}
+	packagePath := path.Join(sublayoutDirectory, "package.2f89b927.link")
+	if err := os.Link("package.2f89b927.link", packagePath); err != nil {
+		t.Errorf("Unable to link package metadata.")
+	}
+
+	var superLayoutMb Metablock
+	if err := superLayoutMb.Load("super.layout"); err != nil {
+		t.Errorf("Unable to load super layout")
 	}
+
```

Comment:
Can you also call `InTotoVerify` on the super layout to test the nesting?

Reply:
And can you also create a test case, where super layout verification fails, because sublayout verification fails?

Reply:
That would require throwing in a bunch of sublayout files into the repository. I was thinking of working on runlib next and at the very least adding signing capabilities so we can generate what we need for the tests. Maybe this can be done then?

Reply:
That of course would be really nice!

Reply:
Okay, I'll place it on the back burner for now then. I think working on runlib will give me a cleaner picture of it too, which is very useful. I suppose we should hold off on merging this till then?